pkg/logs/input/file/scanner.go

```diff
@@ -234,6 +268,65 @@ func (s *Scanner) startNewTailer(file *File, m config.TailingMode) bool {
 	return true
 }
 
+// shouldIgnore resolves symlinks in /var/log/containers in order to use that redirection
+// to validate that we will be reading a file for the correct container.
+func (s *Scanner) shouldIgnore(file *File) bool {
+	infos := make(map[string]string)
+	err := filepath.Walk(ContainersLogsDir, func(containerLogFilename string, info os.FileInfo, err error) error {
+		// we only wants to follow symlinks
+		if info == nil || info.Mode()&os.ModeSymlink != os.ModeSymlink || info.IsDir() {
+			// not a symlink, we are not interested in this file
+			return nil
+		}
+
+		// resolve the symlink
+		podLogFilename, err2 := os.Readlink(containerLogFilename)
+		if err2 != nil {
+			log.Debug("Error while resolving symlink of", containerLogFilename, ":", err)
+			return nil
+		}
+
+		infos[podLogFilename] = containerLogFilename
+		return nil
+	})
+
+	// this is not an error if we we are not currently looking for container logs files,
+	// so not problem and just return false.
+	// Still, we write a debug message to be able to troubleshoot that
+	// in cases we're legitimately looking for containers logs.
+	if err != nil {
+		log.Debug("Can't look for symlinks in /var/log/containers:", err)
+		return false
+	}
+
+	// container id extracted from the file found in /var/log/containers
+	base := filepath.Base(infos[file.Path]) // only the file
+	ext := filepath.Ext(base)               // file extension
+	parts := strings.Split(base, "-")       // get only the container ID part from the file
+	var containerIDFromFilename string
+	if len(parts) > 1 {
+		containerIDFromFilename = strings.TrimSuffix(parts[len(parts)-1], ext)
+	}
+
+	// basic validation of the ID that has been parsed, if it doesn't look like
+	// an ID we don't want to compare another ID to it
+	if containerIDFromFilename == "" || !rxContainerID.Match([]byte(containerIDFromFilename)) {
+		return false
+	}
+
+	if file.Source != nil && file.Source.Config != nil &&
```

Comment:
NIT: you could move this chain of null checks to the beginning of the function to skip all of the logic above in the case that a `Source`, `Source.Config` or `Source.Config.Identifier` doesn't exist. I'm not sure how often this would actually happen in practice though.

Reply:
I'm a big fan of early returns, I'll push that change 👍